### PR TITLE
Refactored authoritative IAM policy logic to be reusable

### PR DESCRIPTION
**Pull request by rosbo**
Similar to the work I did for IAM binding and member resources in #744. 

Could not reuse it for project because of the deprecated `authoritative` boolean field. Once we bump the version and remove the deprecated field in project, we can reuse the same logic. I created an issue to track this work once the field is removed #777.

### CODE REVIEW

**Comment by rosbo** [comment]
```sh
--- PASS: TestAccGoogleFolderIamPolicy_basic (16.82s)
--- PASS: TestAccGoogleFolderIamPolicy_update (20.12s)
```

File: google/iam_folder.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+	return nil
+}
+
+func (u *FolderIamUpdater) GetResourceId() string {
+	return u.folderId
+}
+
+func (u *FolderIamUpdater) GetMutexKey() string {
+	return fmt.Sprintf("iam-project-%s", u.folderId)
```
Comment (danawillow) [inline google/iam_folder.go]: s/project/folder

File: google/iam_folder.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+func (u *FolderIamUpdater) DescribeResource() string {
+	return fmt.Sprintf("project %q", u.folderId)
```
Comment (danawillow) [inline google/iam_folder.go]: here too

File: google/iam_folder.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+	return "folders/" + folder
+}
+
+// v1 and v2beta policy are identical
+func v1PolicyToV2Beta(in *cloudresourcemanager.Policy) (*resourceManagerV2Beta1.Policy, error) {
+	out := &resourceManagerV2Beta1.Policy{}
+	err := Convert(in, out)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot convert a v1 oolicy to a v2beta policy: %s", err)
```
Comment (danawillow) [inline google/iam_folder.go]: typo: s/oolicy/policy

File: google/iam_folder.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	}
+	return out, nil
+}
+
+func v2BetaPolicyToV1(in *resourceManagerV2Beta1.Policy) (*cloudresourcemanager.Policy, error) {
+	out := &cloudresourcemanager.Policy{}
+	err := Convert(in, out)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot convert a v2beta oolicy to a v1 policy: %s", err)
```
Comment (danawillow) [inline google/iam_folder.go]: here too

Comment (rosbo) [reply google/iam_folder.go]: Fixed

File: google/iam_folder.go (hunk 3 above)
Comment (rosbo) [reply google/iam_folder.go]: Fixed

File: google/iam_folder.go (hunk 2 above)
Comment (rosbo) [reply google/iam_folder.go]: Fixed

File: google/iam_folder.go (hunk 1 above)
Comment (rosbo) [reply google/iam_folder.go]: Fixed

**Pull request merged**